### PR TITLE
Refine FeedbackResponseCommentBackend

### PR DESCRIPTION
**Pull request by JosephineKwa**
Style formatted.
Front-end, tests and test coverage will be done in separate PR.

### CODE REVIEW

File: src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java [hunk 1]
```diff
@@ -42,33 +42,34 @@
...
+        try {
             return frcDb.createEntity(frComment);
         } catch (EntityAlreadyExistsException e) {
-            try{
+            try {
                 FeedbackResponseCommentAttributes existingComment = new FeedbackResponseCommentAttributes();
                 
-                existingComment = frcDb.getFeedbackResponseComment(frComment.feedbackResponseId, frComment.giverEmail, frComment.createdAt);
-                if(existingComment == null){
+                existingComment = frcDb.getFeedbackResponseComment(frComment.feedbackResponseId, frComment.giverEmail,
```
Comment (damithc) [inline src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java]: The line needs shortening.

File: src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java [hunk 2]
```diff
@@ -79,22 +80,23 @@ public FeedbackResponseCommentAttributes getFeedbackResponseComment(Long feedbac
...
     }
 
-    public List<FeedbackResponseCommentAttributes> getFeedbackResponseCommentForResponse(String feedbackResponseId){
+    public List<FeedbackResponseCommentAttributes> getFeedbackResponseCommentForResponse(String feedbackResponseId) {
         return frcDb.getFeedbackResponseCommentsForResponse(feedbackResponseId);
     }
     
     public List<FeedbackResponseCommentAttributes> getFeedbackResponseCommentForSession(String courseId,
-            String feedbackSessionName) {
+                                                                                        String feedbackSessionName) {
```
Comment (damithc) [inline src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java]: Can shorten this line? After you create the pull request, view the diff in GitHub to see if any line needs horizontal scrolling to view.

File: src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java [hunk 3]
```diff
@@ -117,31 +120,35 @@ public void updateFeedbackResponseCommentsForResponse(String feedbackResponseId)
...
         
         List<FeedbackResponseCommentAttributes> frcList = new ArrayList<FeedbackResponseCommentAttributes>();
         List<FeedbackSessionAttributes> feedbackSessions = fsLogic.getFeedbackSessionsForCourse(courseId);
-        for(FeedbackSessionAttributes fs:feedbackSessions){
-            if(fs.isPublished()){
-                frcList.addAll(frcDb.getFeedbackResponseCommentsForSendingState(courseId, fs.feedbackSessionName, state));
+        for (FeedbackSessionAttributes fs : feedbackSessions){
+            if (fs.isPublished()) {
+                frcList.addAll(frcDb.getFeedbackResponseCommentsForSendingState(courseId, fs.feedbackSessionName, 
+                                                                                state));
```
Comment (damithc) [inline src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java]:
```
frcList.addAll(frcDb.getFeedbackResponseCommentsForSendingState(
                              courseId, fs.feedbackSessionName, state));
```
Or this one (prefer higher level breaks to lower level breaks)?
```
frcList.addAll(
      frcDb.getFeedbackResponseCommentsForSendingState(courseId, fs.feedbackSessionName, state));
```

File: src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java [hunk 4]
```diff
@@ -269,50 +267,51 @@ public boolean isResponseCommentVisibleForUser(String userEmail, String courseId
...
                         && response.recipientEmail.equals(student.team))
                     || ((relatedQuestion.giverType == FeedbackParticipantType.TEAMS
-                        || isResponseCommentVisibleTo(relatedQuestion, relatedComment, FeedbackParticipantType.OWN_TEAM_MEMBERS))
+                        || isResponseCommentVisibleTo(relatedQuestion, relatedComment,
+                                                      FeedbackParticipantType.OWN_TEAM_MEMBERS))
                             && studentsEmailInTeam.contains(response.giverEmail))
-                    || (isResponseCommentVisibleTo(relatedQuestion, relatedComment, FeedbackParticipantType.RECEIVER_TEAM_MEMBERS)
+                    || (isResponseCommentVisibleTo(relatedQuestion, relatedComment,
+                                                   FeedbackParticipantType.RECEIVER_TEAM_MEMBERS)
                             && studentsEmailInTeam.contains(response.recipientEmail))) {
```
Comment (damithc) [inline src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java]: See if you can extract each condition into an explanatory variable (i.e. apply 'introduce explaining variable' refactoring).

File: src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java [hunk 5]
```diff
@@ -485,11 +490,12 @@ private Object getFeedbackResponseCommentEntity(String courseId, Date createdAt,
         return getCommentsWithoutDeletedEntity(feedbackResponseCommentList);
     }
     
-    private List<FeedbackResponseComment> getFeedbackResponseCommentEntityForSendingState(String courseId, String feedbackSessionName,
-            CommentSendingState state) {
+    private List<FeedbackResponseComment> getFeedbackResponseCommentEntityForSendingState(String courseId,
+            String feedbackSessionName, CommentSendingState state) {
```
Comment (damithc) [inline src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java]: Try to (not compulsory) move the 2nd line to the right of the parent element. It's more intuitive when the child does not appear on the left of the parent element. e.g.
```
private List<FeedbackResponseComment> getFeedbackResponseCommentEntityForSendingState(String courseId,
                                              String feedbackSessionName, CommentSendingState state) {
```

File: src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java [hunk 6]
```diff
@@ -599,21 +607,23 @@ private FeedbackResponseComment getFeedbackResponseCommentEntity(
         @SuppressWarnings("unchecked")
         List<FeedbackResponseComment> firstQueryResponseComments =
             (List<FeedbackResponseComment>) q.execute(courseId, feedbackSessionName, section);
-        for(FeedbackResponseComment responseComment : firstQueryResponseComments){
-            if(!JDOHelper.isDeleted(responseComment)){
-                FeedbackResponseCommentList.put(String.valueOf(responseComment.getFeedbackResponseCommentId()), responseComment);
+        for (FeedbackResponseComment responseComment : firstQueryResponseComments) {
+            if (!JDOHelper.isDeleted(responseComment)) {
+                FeedbackResponseCommentList.put(String.valueOf(responseComment.getFeedbackResponseCommentId()),
+                                                               responseComment);
```
Comment (damithc) [inline src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java]: Break at higher level? i.e. after first `(` ?

**Comment by damithc** [comment]
Added some comments.

File: src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java [hunk 7]
```diff
@@ -586,10 +593,11 @@ private FeedbackResponseComment getFeedbackResponseCommentEntity(
         return getCommentsWithoutDeletedEntity(feedbackResponseCommentList);
     }
     
-    private Collection<FeedbackResponseComment> getFeedbackResponseCommentEntitiesForSessionInSection(
-            String courseId, String feedbackSessionName, String section) {
+    private Collection<FeedbackResponseComment> getFeedbackResponseCommentEntitiesForSessionInSection(String courseId,
+            String feedbackSessionName, String section) {
 
-        Map<String, FeedbackResponseComment> FeedbackResponseCommentList = new HashMap<String, FeedbackResponseComment>();
+        Map<String, FeedbackResponseComment> FeedbackResponseCommentList =
```
Comment (kai33) [inline src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java]: hi can help to change this variable to lower camelCase?

**Comment by JosephineKwa** [comment]
I have renamed the variable to be lower camel case and reformatted the code style where necessary

File: src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java [hunk 8]
```diff
@@ -535,40 +539,42 @@ private FeedbackResponseComment getFeedbackResponseCommentEntity(
...
-            String courseId, String feedbackSessionName) {
+    private List<FeedbackResponseComment> getFeedbackResponseCommentEntitiesForSession(String courseId,
+                                                                                       String feedbackSessionName) {
         
         Query q = getPM().newQuery(FeedbackResponseComment.class);
         q.declareParameters("String courseIdParam, String feedbackSessionNameParam");
-        q.setFilter("courseId == courseIdParam && " +
-                "feedbackSessionName == feedbackSessionNameParam");
+        q.setFilter("courseId == courseIdParam && "
+                + "feedbackSessionName == feedbackSessionNameParam");
```
Comment (damithc) [inline src/main/java/teammates/storage/api/FeedbackResponseCommentsDb.java]: This is fine but even better if you indent it further until it is on the right of `(`

**Comment by damithc** [comment]
@kai33 can you review this one? If busy now, I can assign someone else.

**Comment by kai33** [comment]
sure

**Comment by wkurniawan07** [comment]
@JosephineKwa can you finish and merge this soon? I oversee my #3399 will have huge conflict with this and I'd rather resolve it soon :stuck_out_tongue:

**Comment by JosephineKwa** [comment]
Sure, I will resolve it and hopefully get it approved for merging later tonight!

**Comment by damithc** [comment]
@kai33 for your action please.

**Comment by kai33** [comment]
hi @JosephineKwa have u done the possible improvements that Dr. Damith mentioned?

**Comment by kai33** [comment]
@JosephineKwa 
If done, you can ask @LowWeiLin or @kanghj to merge this tmr. Looks good :+1: 
I'm only available in the evening, may be too late for #3399

**Comment by damithc** [comment]
@kai33 no need to do those. Apparently, we can avoid horizontal scrolling in the diff view by using a smaller font size in the browser. @JosephineKwa can do the merge herself.

**Comment by wkurniawan07** [comment]
Nice, I'll just merge from this branch then!

**Pull request merged**